python/ray/serve/controller.py

```diff
@@ -80,6 +80,74 @@ def __repr__(self) -> str:
         return f"<Traffic {self.traffic_dict}; Shadow {self.shadow_dict}>"
 
 
+class HTTPState:
+    def __init__(self, controller_name: str, detached: bool,
+                 config: HTTPConfig):
+        self.controller_name = controller_name
+        self.detached = detached
+        self.config = config
+        self.proxy_actors: Dict[NodeId, ActorHandle] = dict()
+
+        # Will populate self.proxy_actors with existing actors.
+        self._start_proxies_if_needed()
+
+    def http_proxy_handles(self) -> List[ActorHandle]:
+        return list(self.proxy_actors.values())
+
+    def update(self):
+        self._start_proxies_if_needed()
+        self._stop_proxies_if_needed()
+
+    def _start_proxies_if_needed(self) -> None:
```

Comment:
Could we pull these into a new class like `state.py`?

Reply:
Assume you meant new file. Good idea, I'll do that in a follow-up for this and the EndpointState

Reply:
Oops, yep! Sounds good!